File: 1. Two Sum/twosum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        for (int i = 0; i < nums.size(); i++) {
            int num2 = target - nums[i];
            if (map.find(num2) != map.end()) {
                return {map[num2], i};
            }
            map.insert({nums[i], i});
        }
        return {};
    }
};

File: 1. Two Sum/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for i, num in enumerate(nums):
            num2: int = target - num
            if num2 in dic:
                return [dic[num2], i]
            dic[num] = i